MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift

```diff
@@ -40,63 +37,40 @@ class NewHomeViewController: UIViewController, UITableViewDelegate, UITableViewD
+        }
+
         switch indexPath.section {
         case 0:
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Identifiers.homeScreenTableCell, for: indexPath) as? NewTableViewCell else {
-                return UITableViewCell()
-            }
-            cell.configure(with: nowPlayingViewModel.nowPlayingMovies ?? [], sectionTitle: "  Now Playing")
-            cell.didSelectItem = { [weak self] movie in
-                guard let self = self else { return }
-                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
-            }
-            return cell
+            cell.configure(with: movieViewModel.nowPlayingMovies, sectionTitle: "  Now Playing")
         case 1:
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Identifiers.homeScreenTableCell, for: indexPath) as? NewTableViewCell else {
-                return UITableViewCell()
-            }
-            cell.configure(with: popularMoviesViewModel.popularMovies ?? [], sectionTitle: "  Popular")
-            cell.didSelectItem = { [weak self] movie in
-                guard let self = self else { return }
-                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
-            }
-            return cell
+            cell.configure(with: movieViewModel.popularMovies, sectionTitle: "  Popular")
         case 2:
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Identifiers.homeScreenTableCell, for: indexPath) as? NewTableViewCell else {
-                return UITableViewCell()
-            }
-            cell.configure(with: topRatedMoviesViewModel.topRatedMovies ?? [], sectionTitle: "  Top Rated")
-            cell.didSelectItem = { [weak self] movie in
-                guard let self = self else { return }
-                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
-            }
-            return cell
+            cell.configure(with: movieViewModel.topRatedMovies, sectionTitle: "  Top Rated")
         case 3:
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Identifiers.homeScreenTableCell, for: indexPath) as? NewTableViewCell else {
-                return UITableViewCell()
-            }
-            cell.configure(with: upcomingMoviesViewModel.upcomingMovies ?? [], sectionTitle: "  Upcoming")
-            cell.didSelectItem = { [weak self] movie in
-                guard let self = self else { return }
-                self.performSegue(withIdentifier: Constants.Identifiers.goToMovieDetails, sender: movie.movieID)
-            }
-            return cell
+            cell.configure(with: movieViewModel.upcomingMovies, sectionTitle: "  Upcoming")
         default:
             return UITableViewCell()
         }
+        cell.didSelectItem = { [weak self] movie in
```

Comment:
so you are passing in a closure when the cell is selected to perform a segue
there is a delegate for this. have you tried it?

func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath)

I know you have a mix of a table view and collection view, so it might be tricky. 
try it, and if it doesn't work, let me know.

Reply:
I think I did consider this at first but then realising the movies are individual cells because they are in a collection view in the table view. so the table row must not be selected, but rather the movie cell in the collection view inside the table view.